src/main/kotlin/org/rust/ide/annotator/RsFormatMacroAnnotator.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.ide.annotator.AnnotatorBase
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapiext.isUnitTestMode
+import com.intellij.psi.PsiElement
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.RsLitExpr
+import org.rust.lang.core.psi.RsLiteralKind
+import org.rust.lang.core.psi.RsMacroCall
+import org.rust.lang.core.psi.ext.descendantOfTypeStrict
+import org.rust.lang.core.psi.ext.macroName
+import org.rust.lang.core.psi.ext.startOffset
+import org.rust.lang.core.psi.kind
+import org.rust.lang.utils.parseRustStringCharacters
+
+class RsFormatMacroAnnotator : AnnotatorBase() {
+
+    override fun annotateInternal(element: PsiElement, holder: AnnotationHolder) {
+        val formatMacro = element as? RsMacroCall ?: return
+
+        val formatStr = formatMacro
+            .formatMacroArgument
+            ?.formatMacroArgList
+            ?.getOrNull(macroToFormatPos(formatMacro.macroName))
+            ?.descendantOfTypeStrict<RsLitExpr>()
+            ?: return
+
+        val literalKind = (formatStr.kind as? RsLiteralKind.String) ?: return
+        val rawTextRange = literalKind.offsets.value ?: return
+        val (unescapedText, sourceMap, _) = parseRustStringCharacters(rawTextRange.substring(literalKind.node.text))
+
+        val arguments = formatParser.findAll(unescapedText.toString())
+        var currentOffset = 0
+
+        fun IntRange.toSourceRange(additionalOffset: Int = currentOffset): TextRange =
+            TextRange(sourceMap[first + additionalOffset], sourceMap[last + additionalOffset] + 1)
+                .shiftRight(formatStr.startOffset + rawTextRange.startOffset)
+
+        fun AnnotationHolder.highlightArgument(range: IntRange?, color: RsColor = RsColor.IDENTIFIER) {
+            if (range != null && !range.isEmpty())
+                createAnnotation(
+                    ARG_HIGHLIGHTING,
+                    range.toSourceRange(),
+                    null)
+                    .textAttributes = color.textAttributesKey
+        }
+
+        val key = RsColor.VALID_STRING_ESCAPE
```

Comment:
```suggestion
        val color = RsColor.VALID_STRING_ESCAPE
```

Reply:
Let's use a special color for this just like CLion does for C/C++. Add this to `RsColor`: `FORMAT_SPECIFIER("Literals//Strings//Format specifier in string argument", Default.VALID_STRING_ESCAPE),`